finals-2024-01-15/code/5265-genci-telqiu/src/app/movie-list/movie-list.component.ts

```diff
@@ -1,10 +1,70 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
+import { MovieService } from '../services/movie.service';
 
 @Component({
   selector: 'app-movie-list',
   templateUrl: './movie-list.component.html',
-  styleUrl: './movie-list.component.css'
+  styleUrls: ['./movie-list.component.css'],
 })
-export class MovieListComponent {
+export class MovieListComponent implements OnInit {
+  movies: any[] = [];
+  filteredMovies: any[] = [];
+  genres: string[] = [];
+  titleFilter: string = '';
+  yearFilter: number | null = null;
+  selectedGenre: string | null = null;
+  ratingFilter: number | null = null;
+
+  constructor(private movieService: MovieService) {}
+
+  ngOnInit(): void {
+    this.movieService.getMovies().subscribe((data) => {
+      this.movies = data;
+      this.filteredMovies = this.movies;
+      this.extractGenres();
+    });
+  }
+
+  private extractGenres(): void {
+    const allGenres: string[] = [];
+    this.movies.forEach((movie) => {
+      allGenres.push(...movie.genre);
+    });
+    this.genres = Array.from(new Set(allGenres));
+  }
+
+  sortMoviesBy(property: string): void {
+    this.filteredMovies.sort((a, b) => (a[property] > b[property] ? 1 : -1));
```

Comment:
The sort method is also weird, as it only returns two possibilities, but when we sort we actually have three (bigger, smaller, equal) +3 points